core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -56,16 +58,22 @@ import shapeless.ops.hlist.Tupler
  *   Http.server.serve(foobar.toService)
  * }}}
  */
-trait Endpoint[A] { self =>
+abstract class Endpoint[A] extends Serializable { self =>
 
-  type ContentType <: String
+  /**
+   * The [[StateT]] instance underneath this endpoint.
+   */
+  def run: Endpoint.State[A]
```

Comment:
Yeah, that reminded me fs2 (scalaz-streams) with `.run.run` and I though it would be ok to do the same thing here.

I guess I like `state`.